src/diffusers/schedulers/scheduling_lms_discrete.py

```diff
@@ -215,7 +210,18 @@ def step(
 
         if isinstance(timestep, torch.Tensor):
             timestep = timestep.to(self.timesteps.device)
-        step_index = (self.timesteps == timestep).nonzero().item()
+        if (
+            isinstance(timestep, int)
+            or isinstance(timestep, torch.IntTensor)
+            or isinstance(timestep, torch.LongTensor)
+        ):
+            warnings.warn(
```

Comment:
Please use the deprecate functionality here: https://github.com/huggingface/diffusers/pull/659